test/MaticTokenVesting.test.js

```diff
@@ -0,0 +1,147 @@
+const Token = artifacts.require("./contracts/MaticToken.sol")
+const Vesting = artifacts.require("./contracts/MaticTokenVesting.sol")
+const BigNumber = web3.utils.BigNumber
+
+const toWei = web3.utils.toWei
+async function increaseBlockTime(seconds) {
+  return web3.currentProvider.send(
+    {
+      jsonrpc: "2.0",
+      method: "evm_increaseTime",
+      params: [seconds],
+      id: new Date().getTime()
+    },
+    () => {}
+  )
+}
+async function mineOneBlock() {
+  return web3.currentProvider.send(
+    {
+      jsonrpc: "2.0",
+      method: "evm_mine",
+      id: new Date().getTime()
+    },
+    () => {}
+  )
+}
+
+async function assertRevert(promise) {
+  try {
+    const tx = await promise
+    const receipt = await web3.eth.getTransactionReceipt(tx.tx)
+    if (receipt.gasUsed >= 6700000) {
+      return
+    }
+  } catch (error) {
+    const invalidOpcode = error.message.search("revert") >= 0
+    assert(invalidOpcode, "Expected revert, got '" + error + "' instead")
+    return
+  }
+  assert.ok(false, 'Error containing "revert" must be returned')
+}
+
+require("chai")
+  .use(require("chai-bignumber")(BigNumber))
+  .should()
+
+contract("Token", async accounts => {
+  let owner = accounts[0]
+  let token
+  let vesting
+  let totalSupply
+
+  before(async function() {
+    totalSupply = web3.utils.toBN(10000000000)
+    totalSupply = totalSupply.mul(web3.utils.toBN(10 ** 18))
+    token = await Token.new("Matic test", "MATIC", 18, totalSupply, {
+      from: owner
+    })
+    const total = await token.totalSupply.call()
+    vesting = await Vesting.new(token.address, { from: owner })
```

Comment:
As we discussed, doesn't this need to be in `beforeEach`? - i.e. deploy a fresh instance for every test.

Reply:
I don't think so this is just a single stream of tests.